tensorflow/python/keras/layers/pooling.py

```diff
@@ -586,8 +586,23 @@ class GlobalAveragePooling1D(GlobalPooling1D):
       `(batch_size, features)`
   """
 
-  def call(self, inputs):
-    return backend.mean(inputs, axis=1)
+  def __init__(self, **kwargs):
+    super(GlobalAveragePooling1D, self).__init__(**kwargs)
+    self.supports_masking = True
+
+  def call(self, inputs, mask=None):
+    if mask is not None:
```

Comment:
@martinwicke Hi, I see fchollet has approved the change, do we still need to assign API Review? Moreover, I think v2 golden files are also need to be updated, how do you think?